koku/api/user_access/view.py

```diff
@@ -18,6 +18,36 @@
 
 LOG = logging.getLogger(__name__)
 
+# Backwards Compatability Statement:
+# This endpoint currently supports a type param to show permissions
+# for a single access key. Currently this param has a different
+# return structure than a get on `/user_access/`. However, the
+# type param is used by console dot to show nav links here:
+# https://console.redhat.com/settings/applications/cost-management
+
+# Therefore, we may be able to remove the type param after the transition
+# or atleast modify its return to be the same as when we use `/user_access/`
+# I have marked areas where we are special casing to return the structure
+# console dot expects for now.
+
+ACCESS_KEY_MAPPING = {
+    "aws": ["aws.account", "aws.organizational_unit"],
+    "azure": ["azure.subscription_guid"],
+    "gcp": ["gcp.account", "gcp.project"],
+    "ibm": ["ibm.account"],
+    "oci": ["oci.payer_tenant_id"],
+    "ocp": ["openshift.cluster", "openshift.node", "openshift.project"],
+    "cost_model": ["cost_model"],
+    "settings": ["settings"],
+}
+ACCESS_KEY_MAPPING["any"] = (
+    ACCESS_KEY_MAPPING["aws"]
+    + ACCESS_KEY_MAPPING["azure"]
+    + ACCESS_KEY_MAPPING["gcp"]
+    + ACCESS_KEY_MAPPING["oci"]
+    + ACCESS_KEY_MAPPING["ocp"]
+)
```

Comment:
Modifying a global on every import seems messy. It might be cleaner to change the global to a frozen dataclass instance.

```python
@dataclasses.dataclass
class AccessMapping:
    aws: tuple[str, str] = ("aws.account", "aws.organizational_unit")
    azure: tuple[str] = ("azure.subscription_guid",)
    gcp: tuple[str, str] = ("gcp.account", "gcp.project")
    ibm: tuple[str] = ("ibm.account",)
    oci: tuple[str] = ("oci.payer_tenant_id",)
    ocp: tuple[str, str, str] = ("openshift.cluster", "openshift.node", "openshift.project")
    cost_model: tuple[str] = ("cost_model",)
    settings: tuple[str] = ("settings",)
    any: tuple[str, ...] = None

    def __post_init__(self):
        result = []
        for field in dataclasses.fields(self):
            if field.name == "any":
                continue

            result.extend(getattr(self, field.name))

        self.any = tuple(result)

    def __getitem__(self, item):
        return getattr(self, item)

    def get(self, item, default=None):
        return getattr(self, item, default)

ACCESS_KEY_MAPPING = AccessMapping()
```